plugin/evm/atomic/vm/tx_gossip_test.go

```diff
@@ -0,0 +1,312 @@
+	genesisBytes, err := genesis.MarshalJSON()
+	require.NoError(err)
+
+	responseSender := &enginetest.SenderStub{
+		SentAppResponse: make(chan []byte, 1),
+	}
+	vm := newAtomicTestVM()
+
+	require.NoError(vm.Initialize(
+		ctx,
+		snowCtx,
+		memdb.New(),
+		genesisBytes,
+		nil,
+		nil,
+		nil,
+		responseSender,
+	))
+	require.NoError(vm.SetState(ctx, snow.NormalOp))
+
+	defer func() {
+		require.NoError(vm.Shutdown(ctx))
+	}()
+
+	// sender for the peer requesting gossip from [vm]
+	peerSender := &enginetest.SenderStub{
+		SentAppRequest: make(chan []byte, 1),
+	}
+	network, err := p2p.NewNetwork(logging.NoLog{}, peerSender, prometheus.NewRegistry(), "")
+	require.NoError(err)
+	client := network.NewClient(p2p.AtomicTxGossipHandlerID)
+
+	// we only accept gossip requests from validators
+	requestingNodeID := ids.GenerateTestNodeID()
+	require.NoError(vm.Connected(ctx, requestingNodeID, nil))
+	validatorState.GetCurrentHeightF = func(context.Context) (uint64, error) {
+		return 0, nil
+	}
+	validatorState.GetValidatorSetF = func(context.Context, uint64, ids.ID) (map[ids.NodeID]*validators.GetValidatorOutput, error) {
+		return map[ids.NodeID]*validators.GetValidatorOutput{
+			requestingNodeID: {
+				NodeID: requestingNodeID,
+				Weight: 1,
+			},
+		}, nil
+	}
+
+	// Ask the VM for any new transactions. We should get nothing at first.
+	emptyBloomFilter, err := gossip.NewBloomFilter(
+		prometheus.NewRegistry(),
+		"",
+		config.TxGossipBloomMinTargetElements,
+		config.TxGossipBloomTargetFalsePositiveRate,
+		config.TxGossipBloomResetFalsePositiveRate,
+	)
+	require.NoError(err)
+	emptyBloomFilterBytes, _ := emptyBloomFilter.Marshal()
+	request := &sdk.PullGossipRequest{
+		Filter: emptyBloomFilterBytes,
+		Salt:   agoUtils.RandomBytes(32),
+	}
+
+	requestBytes, err := proto.Marshal(request)
+	require.NoError(err)
+
+	wg := &sync.WaitGroup{}
+	wg.Add(1)
+	onResponse := func(_ context.Context, nodeID ids.NodeID, responseBytes []byte, err error) {
+		require.NoError(err)
+
+		response := &sdk.PullGossipResponse{}
+		require.NoError(proto.Unmarshal(responseBytes, response))
+		require.Empty(response.Gossip)
+		wg.Done()
+	}
+	require.NoError(client.AppRequest(ctx, set.Of(vm.Ctx.NodeID), requestBytes, onResponse))
+	require.NoError(vm.AppRequest(ctx, requestingNodeID, 1, time.Time{}, <-peerSender.SentAppRequest))
+	require.NoError(network.AppResponse(ctx, snowCtx.NodeID, 1, <-responseSender.SentAppResponse))
+	wg.Wait()
+
+	// Issue a tx to the VM
+	utxo, err := addUTXO(
+		memory,
+		snowCtx,
+		ids.GenerateTestID(),
+		0,
+		snowCtx.AVAXAssetID,
+		100_000_000_000,
+		pk.Address(),
+	)
+	require.NoError(err)
+	tx, err := atomic.NewImportTx(vm.Ctx, vm.CurrentRules(), vm.clock.Unix(), vm.Ctx.XChainID, address, vmtest.InitialBaseFee, secp256k1fx.NewKeychain(pk), []*avax.UTXO{utxo})
+	require.NoError(err)
+	require.NoError(vm.AtomicMempool.AddLocalTx(tx))
+
+	// wait so we aren't throttled by the vm
+	time.Sleep(5 * time.Second)
```

Comment:
use `sleepContext` here:
```suggestion
sleepContext(ctx, 5*time.Second)
```